src/main/groovy/com/deploygate/gradle/plugins/tasks/LoginTask.groovy

```diff
@@ -3,33 +3,62 @@ package com.deploygate.gradle.plugins.tasks
 import com.deploygate.gradle.plugins.dsl.DeployGateExtension
 import com.deploygate.gradle.plugins.internal.http.ApiClient
 import com.deploygate.gradle.plugins.internal.http.GetCredentialsResponse
+import com.deploygate.gradle.plugins.tasks.inputs.Credentials
 import com.deploygate.gradle.plugins.utils.BrowserUtils
 import com.deploygate.gradle.plugins.utils.UrlUtils
 import com.sun.net.httpserver.HttpExchange
 import com.sun.net.httpserver.HttpServer
 import org.gradle.api.DefaultTask
 import org.gradle.api.GradleException
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.Optional
 import org.gradle.api.tasks.TaskAction
+import org.jetbrains.annotations.NotNull
+import org.jetbrains.annotations.VisibleForTesting
 
+import javax.inject.Inject
 import java.util.concurrent.CountDownLatch
 import java.util.concurrent.TimeUnit
 import java.util.concurrent.TimeoutException
 
-class LoginTask extends DefaultTask {
+abstract class LoginTask extends DefaultTask {
+
+    @Input
+    @Optional
+    final Property<String> appOwnerName
+
+    @Input
+    @Optional
+    final Property<String> apiToken
 
     @Internal
-    DeployGateExtension deployGateExtension
+    DeployGateExtension deployGateExtension // TODO remove this reference
```

Comment:
Removing this reference may increase the diff by about +500. 😇

Reply:
thank you for care :)